$NetBSD: patch-ae,v 1.34 2017/09/03 11:30:54 leot Exp $

For proper libtool usage split CC into CC_COMPILE and CC with libtool
mode options compile/link respectively.

Same for CXX.

--- Makedefs.in.orig	2017-06-30 15:44:38.000000000 +0000
+++ Makedefs.in
@@ -17,9 +17,11 @@
 
 AR		=	@AR@
 AWK		=	@AWK@
-CC		=	@LIBTOOL@ @CC@
+CC_COMPILE	=	@LIBTOOL@ --mode=compile @CC@
+CC		=	@LIBTOOL@ --mode=link @CC@
 CHMOD		=	@CHMOD@
-CXX		=	@LIBTOOL@ @CXX@
+CXX_COMPILE	=	@LIBTOOL@ --mode=compile @CXX@
+CXX		=	@LIBTOOL@ --mode=link @CXX@
 DSO		=	@DSO@
 DSOXX		=	@DSOXX@
 GZIP		=	@GZIP@
@@ -38,12 +40,12 @@ SHELL		=	/bin/sh
 # Installation programs...
 #
 
-INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@ @INSTALL_STRIP@
-INSTALL_COMPDATA =	$(INSTALL) -c -m 444 @INSTALL_GZIP@
+INSTALL_BIN	=	$(LIBTOOL) --mode=install $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@ @INSTALL_STRIP@
+INSTALL_COMPDATA =	$(INSTALL) -c -m 444
 INSTALL_CONFIG	=	$(INSTALL) -c -m @CUPS_CONFIG_FILE_PERM@
 INSTALL_DATA	=	$(INSTALL) -c -m 444
 INSTALL_DIR	=	$(INSTALL) -d
-INSTALL_LIB	=	$(LIBTOOL) $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@ @INSTALL_STRIP@
+INSTALL_LIB	=	$(LIBTOOL) --mode=install $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@ @INSTALL_STRIP@
 INSTALL_MAN	=	$(INSTALL) -c -m 444
 INSTALL_SCRIPT	=	$(INSTALL) -c -m @CUPS_EXE_FILE_PERM@
 
@@ -141,7 +143,7 @@ CXXLIBS		=	@CXXLIBS@
 DBUS_NOTIFIER	=	@DBUS_NOTIFIER@
 DBUS_NOTIFIERLIBS =	@DBUS_NOTIFIERLIBS@
 DNSSD_BACKEND	=	@DNSSD_BACKEND@
-DSOFLAGS	=	-L../cups @DSOFLAGS@
+DSOFLAGS	=	-L../cups @DSOFLAGS@ $(OPTIM)
 DSOLIBS		=	@DSOLIBS@ $(COMMONLIBS)
 DNSSDLIBS	=	@DNSSDLIBS@
 IPPFIND_BIN	=	@IPPFIND_BIN@
@@ -252,11 +254,11 @@ USBQUIRKS	=	@USBQUIRKS@
 
 .c.o:
 	echo Compiling $<...
-	$(CC) $(ARCHFLAGS) $(OPTIM) $(ALL_CFLAGS) -c -o $@ $<
+	$(CC_COMPILE) $(ARCHFLAGS) $(OPTIM) $(ALL_CFLAGS) -c -o $@ $<
 
 .cxx.o:
 	echo Compiling $<...
-	$(CXX) $(ARCHFLAGS) $(OPTIM) $(ALL_CXXFLAGS) -c -o $@ $<
+	$(CXX_COMPILE) $(ARCHFLAGS) $(OPTIM) $(ALL_CXXFLAGS) -c -o $@ $<
 
 .man.1 .man.1m .man.3 .man.5 .man.7 .man.8:
 	echo Linking $<...
